src/functions/Get-SkipRemainingOnFailurePlugin.ps1

```diff
@@ -0,0 +1,110 @@
+        [Pester.Test]
+        $Test
+    )
+    "Skipped due to previous failure at path:line '$($Test.ExpandedPath):$($Test.StartLine)' and Run.SkipRemainingOnFailure set to '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'"
+}
+
+function Get-SkipRemainingOnFailurePlugin {
+    $p = @{
+        Name = "SkipRemainingOnFailure"
+    }
+
+    if ($PesterPreference.Run.SkipRemainingOnFailure.Value -eq 'Block') {
+        $p.EachTestTeardownEnd = {
+            param($Context)
+
+            # If test is not marked skipped and failed
+            # Go through block tests and child tests and mark unexecuted tests as skipped
+            if (-not $Context.Test.Skipped -and -not $Context.Test.Passed) {
+
+                $errorRecord = [Pester.Factory]::CreateErrorRecord(
+                    'PesterTestSkipped',
+                    (New-SkippedTestMessage -Test $Context.Test),
+                    $null,
+                    $null,
+                    $null,
+                    $false
+                )
+
+                foreach ($test in $Context.Block.Tests) {
+                    if (-not $test.Executed) {
+                        $test.Skip = $true
+                        $test.ErrorRecord.Add($errorRecord)
+                    }
+                }
+
+                foreach ($test in ($Context.Block | View-Flat)) {
+                    if (-not $test.Executed) {
+                        $test.Skip = $true
+                        $test.ErrorRecord.Add($errorRecord)
+                    }
+                }
+            }
+        }
+    }
+
+    elseif ($PesterPreference.Run.SkipRemainingOnFailure.Value -eq 'Container') {
+        $p.EachTestTeardownEnd = {
+            param($Context)
+
+            # If test is not marked skipped and failed
+            # Go through every test in container from block root and marked unexecuted tests as skipped
+            if (-not $Context.Test.Skipped -and -not $Context.Test.Passed) {
+
+                $errorRecord = [Pester.Factory]::CreateErrorRecord(
+                    'PesterTestSkipped',
+                    (New-SkippedTestMessage -Test $Context.Test),
+                    $null,
+                    $null,
+                    $null,
+                    $false
+                )
+
+                foreach ($test in ($Context.Block.Root | View-Flat)) {
+                    if (-not $test.Executed) {
+                        $test.Skip = $true
+                        $test.ErrorRecord.Add($errorRecord)
+                    }
+                }
+            }
+        }
+    }
+
+    elseif ($PesterPreference.Run.SkipRemainingOnFailure.Value -eq 'Run') {
+        $p.EachTestSetupStart = {
+            param($Context)
+
+            # It test has failed at some point during the run
+            # Skip the test before it runs
+            # This handles skipping tests that failed from different containers in the same run
+            if ($script:failedTest) {
```

Comment:
This variable will persist across runs, so the next run will skip all tests. `$script` in Pester-state will be stored as a module variable until Pester is reimported or the variable is overwritten.

I believe this type of run-scoped state is what PluginConfiguration is created for. You can use it like: `$Context.Configuration.SkipRemainingFailedTest = $Context.Test`  +++.

Reply:
Ah yeah forgot about this. I was in the habit of importing before a run so I didn't catch this. Will use pluginconfiguration to store this state instead.